src/iOS/Xamarin.iOS/Samples/Geoprocessing/AnalyzeViewshed/AnalyzeViewshed.cs

```diff
@@ -0,0 +1,212 @@
+            Feature myInputFeature = myInputFeatures.CreateFeature();
+
+            // Assign a physical location to the new point feature based upon where the user clicked in the map view
+            myInputFeature.Geometry = location;
+
+            // Add the new feature with (x,y) location to the feature collection table
+            await myInputFeatures.AddFeatureAsync(myInputFeature);
+
+            // Create the parameters that are passed to the used geoprocessing task
+            GeoprocessingParameters myViewshedParameters =
+                new GeoprocessingParameters(GeoprocessingExecutionType.SynchronousExecute);
+
+            // Request the output features to use the same SpatialReference as the map view
+            myViewshedParameters.OutputSpatialReference = _myMapView.SpatialReference;
+
+            // Add an input location to the geoprocessing parameters
+            myViewshedParameters.Inputs.Add("Input_Observation_Point", new GeoprocessingFeatures(myInputFeatures));
+
+            // Create the job that handles the communication between the application and the geoprocessing task
+            var myViewshedJob = myViewshedTask.CreateJob(myViewshedParameters);
+
+            try
+            {
+                // Execute analysis and wait for the results
+                GeoprocessingResult myAnalysisResult = await myViewshedJob.GetResultAsync();
+
+                // Get the results from the outputs
+                GeoprocessingFeatures myViewshedResultFeatures = myAnalysisResult.Outputs["Viewshed_Result"] as GeoprocessingFeatures;
+
+                // Add all the results as a graphics to the map
+                IFeatureSet myViewshedAreas = myViewshedResultFeatures.Features;
+                foreach (var myFeature in myViewshedAreas)
+                {
+                    _resultOverlay.Graphics.Add(new Graphic(myFeature.Geometry));
+                }
+            }
+            catch (Exception ex)
+            {
+                // Display an error message if there is a problem
+                if (myViewshedJob.Status == JobStatus.Failed && myViewshedJob.Error != null)
+                {
+                    // Report error
+                    UIAlertController alert = UIAlertController.Create("Geoprocessing Error", myViewshedJob.Error.Message, UIAlertControllerStyle.Alert);
+                    alert.AddAction(UIAlertAction.Create("OK", UIAlertActionStyle.Default, null));
+                    PresentViewController(alert, true, null);
+                }
+                else
+                {
+                    // Report error
+                    UIAlertController alert = UIAlertController.Create("Sample Error", ex.ToString(), UIAlertControllerStyle.Alert);
+                    alert.AddAction(UIAlertAction.Create("OK", UIAlertActionStyle.Default, null));
+                    PresentViewController(alert, true, null);
+                }
+            }
+            finally
```

Comment:
I think we should get rid of the finally block. Because we're not doing anything here, I think it might confuse users.